packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -438,6 +438,7 @@ export class ChatContext {
       }
     } catch (e) {
       this._state = priorState;
+      throw e;
     } finally {
       this._batchMode = false;
```

Comment:
Question on catch-finally behavior: will this finally block be called if the exception is rethrown?

Reply:
Yes, finally always runs.